classes/models/component_content.php

```diff
@@ -64,7 +64,7 @@ public function __construct($id, $component, $table, $field, $courseid, $timemod
                                 $title = null, $url = null) {
         parent::__construct($id, $component, $table, $field, $courseid, $timemodified, $contentformat, $title);
         $this->content = $content;
-        $this->contenthash = sha1($this->content);
+        $this->contenthash = is_string($this->content) ? sha1($this->content) : '';
```

Comment:
Have you noticed content not being a string at any point? Is this the fix?

Reply:
You will see failing (core) Behat tests (e.g. admin/tool/uploadcourse/tests/behat/create.future) with PHP 8.1 like this:
`When I click on "Upload courses" "button": debugging() message/s found:
Exception encountered in event observer '\tool_ally\event_handlers::course_created': Unknown error type: sha1(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/admin/tool/ally/classes/models/component_content.php on line 67
`

Reply:
Sorry for the late reply. Since the older PR got merged there will be conflicts with this PR - could you rebase, please?